modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -243,6 +254,40 @@ private void applyLifecycle(List<String> createdComponentModels, boolean applySt
     });
   }
 
+  private BeanDependencyResolver newBeanDependencyResolver() {
+    return beanName -> getDependencyResolver().resolveComponentDependencies(beanName).stream()
+        .map(componentName -> getRegistry().lookupByName(componentName).orElse(null))
+        .filter(component -> component != null)
+        .collect(toList());
+  }
+
+  class ComponentConfigurationLifecycleObjectSorter extends DefaultLifecycleObjectSorter {
+
+    private final BeanDependencyResolver beanDependencyResolver;
+
+    public ComponentConfigurationLifecycleObjectSorter(BeanDependencyResolver beanDependencyResolver) {
+      // Only Object.class type is defined as a single bucket as we just need to sort the configuration components (DSL)
+      super(new Class[] {Object.class});
+      this.beanDependencyResolver = beanDependencyResolver;
+    }
+
+    /**
+     * Implementation that handles duplicates. A->B->C, so if we add B, C will be added but later A will be added and its
```

Comment:
Not clear.

Reply:
Maybe change to something like: 
"Implementation that handles duplicates. 
Whenever a new object is added, then all its dependencies are added as well. So if one of those dependencies wants to be added again, it's ignored."

Reply:
Sure, I forgot to change this javadoc. Makes sense the comment you have suggested.